Jellyfin.Data/Entities/Security/ApiKey.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Globalization;
+
+namespace Jellyfin.Data.Entities.Security
+{
+    /// <summary>
+    /// An entity representing an API key.
+    /// </summary>
+    public class ApiKey
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiKey"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        public ApiKey(string name)
+        {
+            Name = name;
+
+            AccessToken = Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture);
+            DateCreated = DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Gets the id.
+        /// </summary>
+        /// <remarks>
+        /// Identity, Indexed, Required.
+        /// </remarks>
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the date created.
+        /// </summary>
+        public DateTime DateCreated { get; set; }
```

Comment:
Since this entity "lifecycle" is mostly meant to be created and destroyed (i.e not edited, just the name), doesn't make sense that all the properties are set to protected/private by name? What's the purpose of editing the creation/modification date?

This should only be done by EFCore itself in the OnSave override and we shouldn't allow arbitrary modifications to it throughout the code.

Reply:
The setter is public because it needs to be set by the data migration, otherwise I would have it set to private.

Reply:
Can't we make a migrate function that strictly serves for instantiating the entity and setting those properties, so it's only used at migration time?